test/integration/full/preload-cssom/preload-cssom.js

```diff
@@ -0,0 +1,239 @@
+/* global axe, Promise */
+describe('preload cssom integration test pass', function() {
+	'use strict';
+
+	var origAxios;
+
+	before(function(done) {
+		function start() {
+			// Stop messing with my tests Mocha!
+			document.querySelector('#mocha h1').outerHTML =
+				'<h2>preload cssom integration test</h2>';
+			// cache original axios object
+			if (axe.imports.axios) {
+				origAxios = axe.imports.axios;
+			}
+			// run axe
+			axe.run(
+				{
+					preload: true
+				},
+				function(err) {
+					assert.isNull(err);
+					done();
+				}
+			);
+		}
+		if (document.readyState !== 'complete') {
+			window.addEventListener('load', start);
+		} else {
+			start();
+		}
+	});
+
+	function createStub(shouldReject) {
+		/**
+		 * This is a simple override to stub `axe.imports.axios`, until the test-suite is enhanced.
+		 * Did not use any library such as sinon for this, as sinon.stub have difficulties working under selenium webdriver
+		 * Also a generic XHR override was overlooked under webdriver
+		 */
+		axe.imports.axios = function stubbedAxios() {
+			return new Promise(function(resolve, reject) {
+				if (shouldReject) {
+					reject(new Error('Fake Error'));
+				}
+				resolve({
+					data: 'body{overflow:auto;}'
+				});
+			});
+		};
+	}
+
+	function restoreStub() {
+		if (origAxios) {
+			axe.imports.axios = origAxios;
+		}
+	}
+
+	function assertStylesheet(sheet, selectorText, cssText) {
+		assert.isDefined(sheet);
+		assert.property(sheet, 'cssRules');
+		assert.equal(sheet.cssRules[0].selectorText, selectorText);
+		assert.equal(sheet.cssRules[0].cssText.replace(/\s/g, ''), cssText);
+	}
+
+	function getPreload(root) {
+		var config = {
+			asset: 'cssom',
+			timeout: 10000,
+			treeRoot: axe.utils.getFlattenedTree(root ? root : document)
+		};
+		return axe.utils.preloadCssom(config);
+	}
+
+	function commonTestsForRootAndFrame(root) {
```

Comment:
One more thing we forgot to test is if cross domain sheets loaded through @import can be accessed. I suspect they might not, so there's still some work in there I think.

Reply:
Indeed, you are right. I have come up with a solution for the same. Which is why shadowDOM work is going into this PR as against a new one.

Reply:
Note: 
Basic one-level deep nested @import is tackled here.
Have created an issue to tackle further enhancements - https://github.com/dequelabs/axe-core/issues/1054